lib/engine/connection.rb

```diff
@@ -89,16 +89,13 @@ def id
       @id ||=
         begin
           sorted = []
-          path_map = {}
-          node_path = nil
+          junction_map = {}
 
+          # skip over paths that have a junction we've already seen
           @paths.each do |path|
-            node_path = path if path.node?
-            path_map[path] = true
-          end
-
-          node_path.walk(on: path_map) do |path|
-            sorted << path
+            sorted << path unless junction_map[path.a] || junction_map[path.b]
```

Comment:
don't use unless with multiple statements, switch it an if statement

Reply:
Fixed.